active/003-seq-feature.md

```diff
@@ -141,10 +156,18 @@ f.update((1, 2))
 `__del__`
 - Deletes a `BoundFeature`.  The corresponding interval(s) in the `IntervalTree` will be removed when this is called.
 
-## `IntervalMetadata` functions
+
+## `IntervalMetadata` object structure
+### Attributes
+* `features`: a list of `BoundFeature` object
+* `_update_flag`: boolean. Whenever any intervals of any member feature is modified, this is set to True, indicating `_intervaltree` needs to be updated.
+* `_intervaltree` (implemented as property): `IntervalTree` object created from all the intervales in the `features`. When this attributes is accessed by any methods (like `query` below), it checks `_update_flag` to decide whether to re-create from all intervals before it returns the interval tree. This lazy update saves computation.
```

Comment:
This sounds a like a nice place for weird things to happen. But I can't come up with anything unless the `BoundFeature` has a reference to `IntervalTree` which a future maintainer may accidentally access (kind of like how "empty" metadata used to be specially handled before it was removed in https://github.com/biocore/scikit-bio/pull/1321). 

So I guess this will work.